src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,64 @@
+package gift.controller;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import gift.service.ProductService;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+  private final ProductService productService;
+
+  public ProductController(ProductService productService) {
+    this.productService = productService;
```

Comment:
[프로그래밍 요구사항](https://edu.nextstep.camp/s/UH6H8Iba/ls/Mvfn9N7o)에 따르면 아래와 같이 가이드하고있어요. 😃

> 단, 들여쓰기는 '2 spaces'가 아닌 '4 spaces'로 한다.

이런 부분을 일일이 직접 챙기기는 힘들기 때문에 IDE 셋팅으로 들여쓰기를 2 spaces -> 4 spaces로 변경해주신 뒤, 전체적으로 자동정렬( mac 기준: `⌘ + ⌥ + L`, win 기준: `Ctrl + Alt + L`)을 진행해주시면 좋을 것 같아요. 🤗

```suggestion
      this.productService = productService;
```